tests/unit/test_container_to_args_secrets.py

```diff
@@ -89,3 +92,317 @@ async def test_pass_secret_as_env_variable_no_external(self):
+        c.declared_secrets = {
+            "my_secret_name": {
+                "external": "true",
+            }
+        }
+        cnt = get_minimal_container()
+        cnt["_service"] = "test-service"
+        cnt["secrets"] = [
+            {
+                "source": "my_secret_name",
+                "target": "does_not_equal_secret_name",
+                "type": "does_not_equal_env",
+            }
+        ]
+
+        with self.assertRaises(ValueError) as context:
+            await container_to_args(c, cnt)
+        self.assertIn('ERROR: Custom name/target reference ', str(context.exception))
+
+    async def test_secret_target_does_not_match_secret_name_secret_type_env(self):
+        c = create_compose_mock()
+        c.declared_secrets = {
+            "my_secret_name": {
+                "external": "true",
+            }
+        }
+        cnt = get_minimal_container()
+        cnt["_service"] = "test-service"
+        cnt["secrets"] = [
+            {"source": "my_secret_name", "target": "does_not_equal_secret_name", "type": "env"}
+        ]
+
+        args = await container_to_args(c, cnt)
+        self.assertEqual(
+            args,
+            [
+                "--name=project_name_service_name1",
+                "-d",
+                "--network=bridge",
+                "--network-alias=service_name",
+                "--secret",
+                "my_secret_name,type=env,target=does_not_equal_secret_name",
+                "busybox",
+            ],
+        )
+
+    async def test_secret_target_matches_secret_name_secret_type_not_env(self):
+        c = create_compose_mock()
+        c.declared_secrets = {
+            "my_secret_name": {
+                "external": "true",
+            }
+        }
+        cnt = get_minimal_container()
+        cnt["_service"] = "test-service"
+        cnt["secrets"] = [
+            {"source": "my_secret_name", "target": "my_secret_name", "type": "does_not_equal_env"}
+        ]
+
+        with self.assertLogs() as cm:
+            args = await container_to_args(c, cnt)
+        self.assertEqual(len(cm.output), 1)
+        self.assertEqual('That is un-supported and a no-op and is ignored.' in cm.output[0], True)
+        self.assertEqual('my_secret_name' in cm.output[0], True)
+
+        self.assertEqual(
+            args,
+            [
+                "--name=project_name_service_name1",
+                "-d",
+                "--network=bridge",
+                "--network-alias=service_name",
+                "--secret",
+                "my_secret_name,type=does_not_equal_env",
+                "busybox",
+            ],
+        )
+
+    @parameterized.expand([
+        (
+            "no_secret_target",
+            {
+                "file_secret": {
+                    "file": "./my_secret",
+                }
+            },
+            "file_secret",
+            "/test_dirname/my_secret:/run/secrets/file_secret:ro,rprivate,rbind",
```

Comment:
I think we should calculate the final path here, otherwise it's surprising to the reader that the expected value is going to be modified in the test.